internal/controller/bindings/endpointbinding_controller.go

```diff
@@ -213,6 +287,15 @@ func (r *EndpointBindingReconciler) update(ctx context.Context, cr *bindingsv1al
 	var existingTargetService v1.Service
 	var existingUpstreamService v1.Service
 
+	// defer updating statuses
+	defer func() {
+		// best effort
+		_ = r.updateEndpointBindingStatus(ctx, cr)
+
+		// best effort for now TODO(hkatz) retries?
+		_ = postEndpointBindingUpdateToNgrokAPI(ctx, cr)
+	}()
+
```

Comment:
I haven't seen it before. I'm hesitant b/c I think there may be cases where we need to handle these errors  and trigger a re-reconcile instead of just throwing them away. Ex: If we can't update the status in the ngrok API, my 2c is that based on the status code, we re-reconcile again in 30 seconds or so as opposed to waiting the [controller-runtime default of 10 hours](https://github.com/kubernetes-sigs/controller-runtime/blob/3b032e16c0dc19d656626a288cd417d36beaebad/pkg/cache/cache.go#L45) to update it.